fe/fe-core/src/main/java/org/apache/doris/analysis/MultiPartitionDesc.java

```diff
@@ -0,0 +1,316 @@
+                    res = DateTimeFormatter.ofPattern(HOURS_FORMAT);
+                } else if (dateTimeStr.length() == 13) {
+                    res = DateTimeFormatter.ofPattern(HOUR_FORMAT);
+                } else if (dateTimeStr.length() == 19) {
+                    res = DateTimeFormatter.ofPattern(DATETIME_FORMAT);
+                } else {
+                    throw new AnalysisException("can not probe datetime(hour) format:" + dateTimeStr);
+                }
+                break;
+            case DAY: case WEEK:
+                if (dateTimeStr.length() == 8) {
+                    res = DateTimeFormatter.ofPattern(DATES_FORMAT);
+                } else if (dateTimeStr.length() == 10) {
+                    res = DateTimeFormatter.ofPattern(DATE_FORMAT);
+                } else if (dateTimeStr.length() == 19) {
+                    res = DateTimeFormatter.ofPattern(DATETIME_FORMAT);
+                } else {
+                    throw new AnalysisException("can not probe datetime(day or week) format:" + dateTimeStr);
+                }
+                break;
+            case MONTH:
+                if (dateTimeStr.length() == 6) {
+                    res = DateTimeFormatter.ofPattern(MONTHS_FORMAT);
+                } else if (dateTimeStr.length() == 7) {
+                    res = DateTimeFormatter.ofPattern(MONTH_FORMAT);
+                } else if (dateTimeStr.length() == 10) {
+                    res = DateTimeFormatter.ofPattern(DATE_FORMAT);
+                } else if (dateTimeStr.length() == 19) {
+                    res = DateTimeFormatter.ofPattern(DATETIME_FORMAT);
+                } else {
+                    throw new AnalysisException("can not probe datetime(month) format:" + dateTimeStr);
+                }
+                break;
+            case YEAR:
+                if (dateTimeStr.length() == 4) {
+                    res = DateTimeFormatter.ofPattern(YEAR_FORMAT);
+                } else if (dateTimeStr.length() == 8) {
+                    res = DateTimeFormatter.ofPattern(DATES_FORMAT);
+                } else if (dateTimeStr.length() == 10) {
+                    res = DateTimeFormatter.ofPattern(DATE_FORMAT);
+                } else if (dateTimeStr.length() == 19) {
+                    res = DateTimeFormatter.ofPattern(DATETIME_FORMAT);
+                } else {
+                    throw new AnalysisException("can not probe datetime(year) format:" + dateTimeStr);
+                }
+                break;
+            default:
+                throw new AnalysisException("Multi build partition does not support time interval type: "
+                        + timeUnitType);
+        }
+        return res;
+    }
+
+    private DateTimeFormatter dateTypeFormat() {
+        return DateTimeFormatter.ofPattern(this.timeUnitType.equals(TimeUnit.HOUR) ? DATETIME_FORMAT : DATE_FORMAT);
+    }
+
+
+    private List<AllPartitionDesc> aaa(int ii) throws AnalysisException {
```

Comment:
what's this?

Reply:
sorry ，my fault，forgot to delete before submitting.